src/Stryker.Core/Stryker.Core/Reporters/JsonReporter.cs

```diff
@@ -1,82 +1,263 @@
-﻿using Newtonsoft.Json;
+﻿using Microsoft.CodeAnalysis;
+using Newtonsoft.Json;
 using Newtonsoft.Json.Serialization;
 using Stryker.Core.Initialisation.ProjectComponent;
 using Stryker.Core.Mutants;
 using Stryker.Core.Options;
 using System.Collections.Generic;
 using System.IO;
+using System.IO.Abstractions;
 using System.Linq;
 
 namespace Stryker.Core.Reporters
 {
     public class JsonReporter : IReporter
     {
         private readonly StrykerOptions _options;
-        public JsonReporter(StrykerOptions options)
+        private readonly IFileSystem _fileSystem;
+        public JsonReporter(StrykerOptions options, IFileSystem fileSystem)
         {
             _options = options;
+            _fileSystem = fileSystem;
         }
 
         public void OnAllMutantsTested(IReadOnlyInputComponent reportComponent)
         {
-            var jsonReport = JsonReportComponent.FromProjectComponent(reportComponent, _options.ThresholdOptions);
-            jsonReport.Name = _options.ProjectUnderTestNameFilter;
+            var jsonReport = JsonReportComponent.FromProjectComponent(reportComponent, _options);
+            jsonReport.ThresholdHigh = _options.ThresholdOptions.ThresholdHigh;
+            jsonReport.ThresholdLow = _options.ThresholdOptions.ThresholdLow;
+            jsonReport.ThresholdBreak = _options.ThresholdOptions.ThresholdBreak;
 
-            using (var file = File.CreateText(_options.BasePath + "/StrykerLogs/mutation-report.json"))
-            {
-                file.WriteLine(SerializeJsonReport(jsonReport));
-            }
+            WriteReportToJsonFile(jsonReport, Path.Combine(_options.BasePath, "/StrykerLogs/mutation-report.json"));
         }
 
-        private static string SerializeJsonReport(JsonReportComponent jsonReport)
+        private void WriteReportToJsonFile(JsonReportComponent jsonReport, string filePath)
         {
-            return JsonConvert.SerializeObject(jsonReport, new JsonSerializerSettings
+            var json = JsonConvert.SerializeObject(jsonReport, new JsonSerializerSettings
             {
                 ContractResolver = new DefaultContractResolver
                 {
                     NamingStrategy = new CamelCaseNamingStrategy()
                 },
-                Formatting = Formatting.Indented
+                Formatting = Formatting.Indented,
+                NullValueHandling = NullValueHandling.Ignore
             });
+
+            _fileSystem.Directory.CreateDirectory(Path.GetDirectoryName(filePath));
+            using (var file = _fileSystem.File.CreateText(filePath))
+            {
+                file.WriteLine(json);
+            }
         }
 
         public sealed class JsonReportComponent
```

Comment:
Can this class be moved to another file?

Reply:
For now the only purpose of this class is to be able to create the report. It is not used anywhere else and does not provide anything useful to the rest of the project. Why would you like to see it moved?